src/Powercord/plugins/pc-announcements/index.js

```diff
@@ -50,15 +53,18 @@ module.exports = class Announcements extends Plugin {
   }
 
   sendNotice (notice, alwaysDisplay) {
-    if (!this.notices.find(n => n.id === notice.id) && (alwaysDisplay || !this.settings.get('dismissed', []).includes(notice.id))) {
+    notice.alwaysDisplay = alwaysDisplay || false; // append a new key to the corresponding notice; we'll use this to grab the value of alwaysDisplay outside of this method
+    if (!this.notices.find(n => n.id === notice.id) && (notice.alwaysDisplay || !this.settings.get('dismissed', []).includes(notice.id))) {
       this.notices.push(notice);
       this._renderNotice();
     }
   }
 
   closeNotice (noticeId) {
     this.notices = this.notices.filter(n => n.id !== noticeId);
-    this.settings.set('dismissed', [ ...this.settings.get('dismissed', []), noticeId ]);
+    if (this.notices.find(n => n.alwaysDisplay)) { // make sure that we're only adding notices found without alwaysDisplay to the 'dismissed' array
```

Comment:
this will prevent *any* notice to be added to the dismissed store if any announcement have alwaysDisplay on, even if the notice don't have it. You need to grab the notice using its ID (before we remove it from the store), and then check for the value.